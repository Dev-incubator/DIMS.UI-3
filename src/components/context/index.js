import React from 'react';

export const { Provider: ThemeContextProvider, Consumer: ThemeContextConsumer } = React.createContext();
export const { Provider: RoleContextProvider, Consumer: RoleContextCosumer } = React.createContext();
export const { Provider: FetchServiceProvider, Consumer: FetchServiceCosumer } = React.createContext();
