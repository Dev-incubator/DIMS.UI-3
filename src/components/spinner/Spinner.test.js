import React from 'react';
import { configure, shallow, mount } from 'enzyme';
import Adaptor from 'enzyme-adapter-react-16';
import toJson from 'enzyme-to-json';
import Spinner from './Spinner';
import { ThemeContextProvider } from '../context';

configure({
  adapter: new Adaptor(),
});

describe('<Spinner/>', () => {
  it('Snapshot Spinner component', () => {
    const contextThemeValue = { theme: 'dark' };
    const wrapper = mount(
      <ThemeContextProvider value={contextThemeValue}>
        <Spinner />
      </ThemeContextProvider>,
    );
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
