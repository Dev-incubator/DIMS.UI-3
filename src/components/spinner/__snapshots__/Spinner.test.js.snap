// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Spinner/> Snapshot Spinner component 1`] = `
<Component>
  <Spinner
    theme="dark"
  >
    <div
      className="loadingio-spinner-ripple dark"
    >
      <div
        className="ldio"
      >
        <div />
        <div />
      </div>
    </div>
  </Spinner>
</Component>
`;
